frontend/src/metabase/collections/containers/FormCollectionPicker/FormCollectionPicker.tsx

```diff
@@ -107,7 +108,8 @@ function FormCollectionPicker({
 
   const isOpenCollectionInPersonalCollection = openCollection?.is_personal;
   const showCreateNewCollectionOption =
-    !showOnlyPersonalCollections || isOpenCollectionInPersonalCollection;
+    filterPersonalCollections !== "only" ||
```

Comment:
The previous condition was `!showOnlyPersonalCollections`

Given `showOnlyPersonalCollections`  === `filterPersonalCollections === "only"`

that means `!showOnlyPersonalCollections` === `filterPersonalCollections !== "only"`